src/orion/algo/hyperband.py

```diff
@@ -202,13 +247,63 @@ def suggest(self, num=1):
         # complete
         return None
 
-    def get_id(self, point):
-        """Compute a unique hash for a point based on params, but not fidelity level."""
+    def get_id(self, point, ignore_fidelity=True):
+        """Compute a unique hash for a point based on params, without fidelity level by default."""
         _point = list(point)
-        non_fidelity_dims = _point[0:self.fidelity_index]
-        non_fidelity_dims.extend(_point[self.fidelity_index + 1:])
+        if ignore_fidelity:
+            non_fidelity_dims = _point[0:self.fidelity_index]
+            non_fidelity_dims.extend(_point[self.fidelity_index + 1:])
+            _point = non_fidelity_dims
+
+        return hashlib.md5(str(_point).encode('utf-8')).hexdigest()
+
+    def _refresh_bracket(self):
+        """Refresh bracket if one hyperband execution is done"""
+        if all(bracket.is_done for bracket in self.brackets):
+            self.executed_times = self.brackets[0].frequency_id
+            logger.debug('Hyperband execution %i is done, required to execute %s times',
+                         self.executed_times, str(self.frequency))
+
+            # Continue to the next execution if need
+            if self.executed_times < self.frequency:
+                self.brackets = [
+                    Bracket(self, bracket_budgets, self.executed_times + 1)
+                    for bracket_budgets in self.budgets
+                ]
+                if self.seed is not None:
+                    self.seed += 1
+
+    def _get_bracket(self, point):
+        """Get the bracket of a point during observe"""
+        fidelity = point[self.fidelity_index]
+        _id_wo_fidelity = self.get_id(point)
+
+        brackets = []
+        for bracket in self.brackets:
+            # If find same point in first rung of a bracket,
+            # the point should register in this bracket
+            if _id_wo_fidelity in bracket.rungs[0]['results']:
+                brackets = [bracket]
+                break
+
+        if not brackets:
+            # If the point show in current hyeprband execution the first time,
+            # the bracket with same fidelity in the first rung should be used,
+            # the assumption is that there is no duplicated points inside same hyperband execution.
+            brackets = [bracket for bracket in self.brackets
+                        if bracket.rungs[0]['resources'] == fidelity]
+
+        if not brackets:
+            raise ValueError("No bracket found for point {0} with fidelity {1}"
+                             .format(_id_wo_fidelity, fidelity))
 
-        return hashlib.md5(str(non_fidelity_dims).encode('utf-8')).hexdigest()
+        if len(brackets) > 1:
+            logger.warning("More than one bracket found for point %s, this should not happen",
```

Comment:
This could happen now with repetitions as described here  https://github.com/Epistimio/orion/pull/363/files#diff-fd0f6183ef73c491312c6e13328f8353R169, right?

Reply:
I think it should not, if there are more than bracket found for a point, means there are duplicate points in the same hyeprband execution, which is not allowed. We only allow duplicate in different hyperband execution, which should not cause more brackets found a point as the self.brackets here only have bracket for the current hyperband exeucton.

Reply:
Oh! I didn't realize that self.brackets was replaced here: https://github.com/Epistimio/orion/pull/363/files#diff-fd0f6183ef73c491312c6e13328f8353R269. I thought you were appending them. Gotcha, thanks for the explanation!